pkg/controller/machine.go

```diff
@@ -428,6 +429,27 @@ func (c *controller) machineCreate(machine *v1alpha1.Machine, driver driver.Driv
 			klog.Warning(err)
 		}
 
+		// Check if the creation-error logs contain the specific pattern, and annotate the machine-deployment if found.
+		contains, _ := c.containsSpecialPattern(err.Error(), MachineTypeNotAvailableAzure, MachineTypeNotAvailableAWS)
+
+		if contains {
+			md := c.getMachineDeploymentForMachine(machine)
+			if md != nil {
+				mdCopy := md.DeepCopy()
+				annotations := mdCopy.Spec.Template.Spec.NodeTemplateSpec.Annotations
+				if annotations == nil {
+					annotations = make(map[string]string)
+				}
+				annotations[MachineTypeNotAvailableAnnotation] = "True"
+				mdCopy.Spec.Template.Spec.NodeTemplateSpec.Annotations = annotations
+
+				_, errUpdate := c.controlMachineClient.MachineDeployments(machine.Namespace).Update(mdCopy)
+				if errUpdate != nil {
+					klog.Errorf("Error updating the machine-deployment %+v", errUpdate)
```

Comment:
Not returning the error here, to keep the error the intact from the cloud-provider, or should we return the errUpdate ?